libs/authentication/api/src/lib/session.strategy.ts

```diff
@@ -0,0 +1,24 @@
+import {Injectable} from '@nestjs/common'
+import {PassportStrategy} from '@nestjs/passport'
+import {Strategy} from 'passport-http-bearer'
+import {AuthenticationService} from './authentication.service'
+
+@Injectable()
+export class SessionStrategy extends PassportStrategy(Strategy, 'session') {
+  constructor(private authService: AuthenticationService) {
+    super({
+      session: false // would use a cookie if set to true
+    })
+  }
+
+  public async validate(token: string) {
+    const session = await this.authService.getSessionByToken(token)
+    const isSessionValid = this.authService.isSessionValid(session)
+
+    if (!isSessionValid) {
+      return false
```

Comment:
Is this throwing the `UnauthorizedException`?

Reply:
Also, yes